x-pack/plugins/ml/server/models/results_service/build_anomaly_table_items.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import _ from 'lodash';
+import moment from 'moment';
+
+import {
+  getEntityFieldName,
+  getEntityFieldValue,
+  showActualForFunction,
+  showTypicalForFunction
+} from '../../../common/util/anomaly_utils';
+
+
+// Builds the items for display in the anomalies table from the supplied list of anomaly records.
+export function buildAnomalyTableItems(anomalyRecords, aggregationInterval) {
+
+  // Aggregate the anomaly records if necessary, and create skeleton display records with
+  // time, detector (description) and source record properties set.
+  let displayRecords = [];
+  if (aggregationInterval !== 'second') {
+    displayRecords = aggregateAnomalies(anomalyRecords, aggregationInterval);
+  } else {
+    // Show all anomaly records.
+    displayRecords = anomalyRecords.map((record) => {
+      return {
+        time: record.timestamp,
+        source: record,
+      };
+    });
+  }
+
+  // Fill out the remaining properties in each display record
+  // for the columns to be displayed in the table.
+  displayRecords.forEach((record) => {
```

Comment:
nit: this could be refactored to use `.map()` instead of `.forEach()` so it's more clear that we're just augmenting the existing `displayRecords` and not any additional data outside the callback.